app/src/main/java/at/bitfire/icsdroid/ui/SubscriptionSettingsFragment.kt

```diff
@@ -5,157 +5,206 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.Card
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.OutlinedTextField
+import androidx.compose.material.Text
+import androidx.compose.material.TextField
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.rounded.Circle
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.unit.dp
 import androidx.fragment.app.Fragment
 import androidx.fragment.app.activityViewModels
 import androidx.lifecycle.MutableLiveData
-import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModel
 import at.bitfire.icsdroid.R
-import at.bitfire.icsdroid.databinding.SubscriptionSettingsBinding
-import com.google.android.material.dialog.MaterialAlertDialogBuilder
-import com.google.android.material.switchmaterial.SwitchMaterial
-import org.joda.time.Minutes
-import org.joda.time.format.PeriodFormat
+import at.bitfire.icsdroid.ui.reusable.SwitchSetting
 
 class SubscriptionSettingsFragment : Fragment() {
 
     private val model by activityViewModels<SubscriptionSettingsModel>()
 
-    private lateinit var binding: SubscriptionSettingsBinding
-
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, inState: Bundle?): View {
-        binding = SubscriptionSettingsBinding.inflate(inflater, container, false)
-        binding.lifecycleOwner = this
-        binding.model = model
-
-        model.defaultAlarmMinutes.observe(
-            viewLifecycleOwner,
-            defaultAlarmObserver(
-                binding.defaultAlarmSwitch,
-                binding.defaultAlarmText,
-                model.defaultAlarmMinutes
-            )
-        )
-        model.defaultAllDayAlarmMinutes.observe(
-            viewLifecycleOwner,
-            defaultAlarmObserver(
-                binding.defaultAlarmAllDaySwitch,
-                binding.defaultAlarmAllDayText,
-                model.defaultAllDayAlarmMinutes
-            )
-        )
-
         val colorPickerContract = registerForActivityResult(ColorPickerActivity.Contract()) { color ->
             model.color.value = color
         }
-        binding.color.setOnClickListener {
-            colorPickerContract.launch(model.color.value)
+        return ComposeView(requireActivity()).apply {
+            setContent {
+                SubscriptionSettingsComposable(
+                    url = model.url.observeAsState("").value,
```

Comment:
I'd also use here the `by` operator. Ex:
```kotlin
val url by model.url.observeAsState("")
...
```

Reply:
yup, kinda irritating, but better be consistent